app/models/nic/base.rb

```diff
@@ -142,30 +142,28 @@ def host_managed?
       self.host && self.host.managed? && SETTINGS[:unattended]
     end
 
-    def require_ip_validation?(ip_field, other_ip_field, subnet_object)
-      raise ::Foreman::Exception.new("invalid ip field: %s", ip_field) unless ip_field == :ip || ip_field == :ip6
-      raise ::Foreman::Exception.new("invalid other ip field: %s", other_ip_field) unless other_ip_field == :ip || other_ip_field == :ip6
-
-      # if it's not managed there's nowhere to specify an IP anyway
-      return false unless self.host.managed? && self.managed? && self.provision?
-
-      # if the CR will provide an IP, then don't validate yet
-      return false if host.compute_provides?(ip_field)
+    def require_ip4_validation?(from_compute = true)
+      (@require_ip4_validation ||= NicIpRequired.new(:ip, :nic => self, :from_compute => from_compute)).required?
```

Comment:
`NicIpRequired::Ip4.new` would be much simpler (or `Ipv4` rather than `Ip4` for consistency with Subnet)